### DISCUSSION TITLE
Type Definition File for ExtendedHostFunctions

### DISCUSSION THREAD

**Discussion by geometry** [ClearFoundry/ClearScript#258] [Q&A]
Hi,
Does anyone have an ExtendedHostFunctions.d.ts file that they could share?
Looking to generate one from the code: tried several automated tools, but no good results so far.
Cheers.

**Comment by ClearScriptLib** [comment]
Hi @geometry,

Unfortunately we don't have TypeScript definition files for any part of ClearScript's API, and we suspect that tools without specific ClearScript support are unlikely to generate good results due to ClearScript's unique handling of generic type arguments.

Sorry!
